### ISSUE TITLE
Matplotlib version issue

### ISSUE THREAD

**Issue by ckinzthompson** [GonzalezBiophysicsLab/tmaven#5]
Running tmaven using matplotlib 3.8.0 gives the following error:

``` bash
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/colin/programs/tmaven/tmaven/__main__.py", line 8, in <module>
    main()
  File "/home/colin/programs/tmaven/tmaven/__main__.py", line 5, in main
    run_app(sys.argv)
  File "/home/colin/programs/tmaven/tmaven/app.py", line 65, in run_app
    gui.restore_session()
  File "/home/colin/programs/tmaven/tmaven/interface/main_window.py", line 490, in restore_session
    self.plot_container.change_mode(plot_mode)
  File "/home/colin/programs/tmaven/tmaven/trace_plot/plot_container.py", line 23, in change_mode
    self.change_plotter(fret_canvas)
  File "/home/colin/programs/tmaven/tmaven/trace_plot/plot_container.py", line 49, in change_plotter
    temp_plot = canvas(self.gui)
  File "/home/colin/programs/tmaven/tmaven/trace_plot/fret_plot.py", line 68, in __init__
    self.initialize_plots()
  File "/home/colin/programs/tmaven/tmaven/trace_plot/fret_plot.py", line 183, in initialize_plots
    self.ax[0][0].get_shared_y_axes().join(self.ax[0][0],self.ax[0][1])
AttributeError: 'GrouperView' object has no attribute 'join'. Did you mean: 'joined'?
```

This is fixed by downgrading to 3.7 or lower. The issue is in how we make the histogram and the trace plots have the same axis

**Comment by ckinzthompson** [comment]
It looks like the issue is how matplotlib 3.8+ handles `.get_shared_y_axes()`. It now returns a `GrouperView` which is an immutable look at a `Grouper` -- (see here)[https://github.com/matplotlib/matplotlib/blob/c5707d9c797e2a04490f2ae5bab0e4d19f8cbc51/lib/matplotlib/cbook.py#L815C1-L815C14]. `Grouper` does have `.join()` but `GrouperView` only has `.joined()` which checks if two things are joined in the `Grouper`.

**Comment by ckinzthompson** [comment]
fixed in dev 0.2.1 - being pushed to main soon
